circuits/src/withdraw.rs

```diff
@@ -0,0 +1,182 @@
+    r: Vec<Vec<String>>,
+    s: Vec<Vec<String>>,
+    msghash: Vec<Vec<String>>,
+    pubkey: Vec<[Vec<String>; 2]>,
+}
+
+pub fn nova(iteration_count: usize) {
+    let root = current_dir().unwrap();
+
+    let circuit_file = root.join("circuits/build/compiled_circuit/withdraw.r1cs");
+    let r1cs = load_r1cs(&FileLocation::PathBuf(circuit_file));
+    let witness_generator_wasm = root.join("circuits/build/compiled_circuit/withdraw_js/withdraw.wasm");
+
+    let withdraw_data: Withdraw = serde_json::from_str(include_str!("../inputs/withdraw.json")).unwrap();
+
+    println!("json: {:?}", withdraw_data);
+
+    let start_public_input = vec![
+        F1::from_str_vartime(&withdraw_data.step_in[0]).unwrap(),
+    ];
+
+    let mut private_inputs = Vec::new();
+
+    for i in 0..iteration_count {
+        let mut private_input = HashMap::new();
+        private_input.insert(
+            "sender".to_string(),
+            json!(withdraw_data.sender[i]),
+        );
+        private_input.insert(
+            "recipient".to_string(),
+            json!(withdraw_data.recipient[i]),
+        );
+        private_input.insert(
+            "leaf_coins".to_string(),
+            json!(withdraw_data.leaf_coins[i]),
+        );
+        private_input.insert(
+            "pathElements".to_string(),
+            json!(withdraw_data.pathElements[i]),
+        );
+        private_input.insert(
+            "pathIndices".to_string(),
+            json!(withdraw_data.pathIndices[i]),
+        );
+        private_input.insert(
+            "r".to_string(),
+            json!(withdraw_data.r[i]),
+        );
+        private_input.insert(
+            "s".to_string(),
+            json!(withdraw_data.s[i]),
+        );
+        private_input.insert(
+            "msghash".to_string(),
+            json!(withdraw_data.msghash[i]),
+        );
+        private_input.insert(
+            "pubkey".to_string(),
+            json!(withdraw_data.pubkey[i]),
+        );
+
+        private_inputs.push(private_input);
+    }
+    dbg!(&private_inputs);
+
+
+    let pp = create_public_params(r1cs.clone());
+
+    println!(
+        "Number of constraints per step (primary circuit): {}",
+        pp.num_constraints().0
+    );
+    println!(
+        "Number of constraints per step (secondary circuit): {}",
+        pp.num_constraints().1
+    );
+
+    println!(
+        "Number of variables per step (primary circuit): {}",
+        pp.num_variables().0
+    );
+    println!(
+        "Number of variables per step (secondary circuit): {}",
+        pp.num_variables().1
+    );
+
+    println!("Creating a RecursiveSNARK...");
+    let start = Instant::now();
+    let recursive_snark = create_recursive_circuit(
+        FileLocation::PathBuf(witness_generator_wasm),
+        r1cs,
+        private_inputs,
+        start_public_input.clone(),
+        &pp,
+    )
+    .unwrap();
+    println!("RecursiveSNARK creation took {:?}", start.elapsed());
+
+    // TODO: empty?
```

Comment:
I know this is just copied from nova-scotia repo, but it makes me nervous, since idk what it means 😅 
Any ideas? We could ask on the upstream repo?

Reply:
no idea! we should ask.